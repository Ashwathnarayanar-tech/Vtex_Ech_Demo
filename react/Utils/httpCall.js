export async function makeAPICall(appURL, method, reqRange, reqData) {
  if (method == "GET") {
    try {
      let header = new Headers();
      header.append("Content-Type", "application/json");
      if (reqRange) {
        header.append("REST-Range", `resources=${reqRange}`);
      }
      header.append("Access-Control-Allow-Methods", "GET, OPTIONS, POST, PUT");
      const response = await fetch(appURL, {
        mode: "cors",
        credentials: "include",
        method: method,
        headers: header,
      });
      let data = await response.json();
      //   console.log({ data });
      return data;
    } catch (e) {
      console.log(e);
    }
  } else if (method == "POST") {
  } else {
    /* Do Nothing */
  }
}
