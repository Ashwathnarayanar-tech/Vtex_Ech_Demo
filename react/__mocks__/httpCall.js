import fetch from "node-fetch";

export async function makeAPICall(method) {
        let header = new Headers();
        header.append("Content-Type", "application/json");
        header.append("Access-Control-Allow-Methods", "GET");
        return fetch("https://echidna.vtexcommercestable.com.br/api/catalog_system/pub/category/tree/10", {
          mode: "cors",
          credentials: "include",
          method: method,
          headers: header,
        }
      
        )
        .then((response)=>{
            return response.json();
        })
    }