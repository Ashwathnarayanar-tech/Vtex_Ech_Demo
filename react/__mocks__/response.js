export const expectedResponse =     

[
  {
    "id": 1,
    "name": "Test Category",
    "hasChildren": true,
    "url": "https://echidna.vtexcommercestable.com.br/test-category",
    "children": [],
    "Title": "Category_Page_Title",
    "MetaTagDescription": "Brief description of the category. It is advisable not to exceed 150 characters so that the search engines can show it correctly in the search results;"
  },
  {
    "id": 101,
    "name": "Major Appliances",
    "hasChildren": true,
    "url": "https://echidna.vtexcommercestable.com.br/major-appliances",
    "children": [
      {
        "id": 104,
        "name": "Refrigerators",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/major-appliances/refrigerators",
        "children": [],
        "Title": "Refrigerators",
        "MetaTagDescription": "This is a generic description of a Department."
      },
      {
        "id": 105,
        "name": "Ranges & Ovens",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/major-appliances/ranges---ovens",
        "children": [],
        "Title": "Ranges & Ovens",
        "MetaTagDescription": "This is a generic description of a Department."
      },
      {
        "id": 106,
        "name": "Washers & Dryers",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/major-appliances/washers---dryers",
        "children": [],
        "Title": "Washers & Dryers",
        "MetaTagDescription": "This is a generic description of a Category."
      }
    ],
    "Title": "Major Appliances",
    "MetaTagDescription": "This is a generic description of a Department."
  },
  {
    "id": 102,
    "name": "Small Appliances",
    "hasChildren": true,
    "url": "https://echidna.vtexcommercestable.com.br/small-appliances",
    "children": [
      {
        "id": 107,
        "name": "Mixers & Blenders",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/small-appliances/mixers---blenders",
        "children": [],
        "Title": "Mixers & Blender",
        "MetaTagDescription": "This is a generic description of a Category."
      },
      {
        "id": 108,
        "name": "Toasters & Grills",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/small-appliances/toasters---grills",
        "children": [],
        "Title": "Toasters & Grills",
        "MetaTagDescription": "This is a generic description of a Category."
      },
      {
        "id": 109,
        "name": "Coffee Makers",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/small-appliances/coffee-makers",
        "children": [],
        "Title": "Coffee Makers",
        "MetaTagDescription": "This is a generic description of a Category."
      },
      {
        "id": 120,
        "name": "Test-Category",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/small-appliances/test-category",
        "children": [],
        "Title": "test",
        "MetaTagDescription": ""
      }
    ],
    "Title": "Small Appliances",
    "MetaTagDescription": "This is a generic description of a Department."
  },
  {
    "id": 128,
    "name": "NISMO",
    "hasChildren": true,
    "url": "https://echidna.vtexcommercestable.com.br/nismo",
    "children": [
      {
        "id": 155,
        "name": "Apparel",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/apparel",
        "children": [],
        "Title": "Apparel",
        "MetaTagDescription": "Apparel"
      },
      {
        "id": 156,
        "name": "Brakes",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/brakes",
        "children": [],
        "Title": "Brakes",
        "MetaTagDescription": "Brakes"
      },
      {
        "id": 157,
        "name": "Chassis",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/chassis",
        "children": [],
        "Title": "Chassis",
        "MetaTagDescription": "Chassis"
      },
      {
        "id": 158,
        "name": "Electrical",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/electrical",
        "children": [],
        "Title": "Electrical",
        "MetaTagDescription": "Electrical"
      },
      {
        "id": 159,
        "name": "Engine Components",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/engine-components",
        "children": [],
        "Title": "Engine-Components",
        "MetaTagDescription": "Engine Components"
      },
      {
        "id": 160,
        "name": "Exterior",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/exterior",
        "children": [],
        "Title": "Exterior",
        "MetaTagDescription": "Exterior"
      },
      {
        "id": 161,
        "name": "Exterior-Emblems",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/exterior-emblems",
        "children": [],
        "Title": "Exterior-Emblems",
        "MetaTagDescription": "Exterior-Emblems"
      },
      {
        "id": 162,
        "name": "Interior",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/interior",
        "children": [],
        "Title": "Interior",
        "MetaTagDescription": "Interior"
      },
      {
        "id": 163,
        "name": "Merchandise",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/merchandise",
        "children": [],
        "Title": "Merchandise",
        "MetaTagDescription": "Merchandise"
      },
      {
        "id": 164,
        "name": "Nismo Accessories",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/nismo-accessories",
        "children": [],
        "Title": "Nismo-Accessories",
        "MetaTagDescription": "Nismo Accessories"
      },
      {
        "id": 165,
        "name": "Power Train-Final Drive",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/power-train-final-drive",
        "children": [],
        "Title": "Power-Train-Final-Drive",
        "MetaTagDescription": "Power Train-Final Drive"
      },
      {
        "id": 166,
        "name": "Suspension",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/suspension",
        "children": [],
        "Title": "Suspension",
        "MetaTagDescription": "Suspension"
      },
      {
        "id": 174,
        "name": "Wheels",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/nismo/wheels",
        "children": [],
        "Title": "Wheels",
        "MetaTagDescription": "Wheels"
      }
    ],
    "Title": "NISMO",
    "MetaTagDescription": "NISMO"
  },
  {
    "id": 103,
    "name": "Electronics",
    "hasChildren": true,
    "url": "https://echidna.vtexcommercestable.com.br/electronics",
    "children": [
      {
        "id": 110,
        "name": "TVs",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/electronics/tvs",
        "children": [],
        "Title": "TVs",
        "MetaTagDescription": "This is a generic description of a Category."
      },
      {
        "id": 111,
        "name": "Laptops",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/electronics/laptops",
        "children": [],
        "Title": "Laptops",
        "MetaTagDescription": "This is a generic description of a Category."
      },
      {
        "id": 112,
        "name": "Cameras",
        "hasChildren": true,
        "url": "https://echidna.vtexcommercestable.com.br/electronics/cameras",
        "children": [],
        "Title": "Cameras",
        "MetaTagDescription": "This is a generic description of a Category."
      },
      {
        "id": 119,
        "name": "Headphones",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/electronics/headphones",
        "children": [],
        "Title": "",
        "MetaTagDescription": ""
      }
    ],
    "Title": "Electronics",
    "MetaTagDescription": "This is a generic description of a Department."
  },
  {
    "id": 125,
    "name": "TRUCKS",
    "hasChildren": true,
    "url": "https://echidna.vtexcommercestable.com.br/trucks",
    "children": [
      {
        "id": 126,
        "name": "Frontier",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/trucks/frontier",
        "children": [],
        "Title": "Frontier",
        "MetaTagDescription": ""
      },
      {
        "id": 127,
        "name": "Titan",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/trucks/titan",
        "children": [],
        "Title": "Titan",
        "MetaTagDescription": ""
      }
    ],
    "Title": "TRUCKS",
    "MetaTagDescription": ""
  },
  {
    "id": 129,
    "name": "CROSSOVERS & SUVS",
    "hasChildren": true,
    "url": "https://echidna.vtexcommercestable.com.br/crossovers---suvs",
    "children": [
      {
        "id": 130,
        "name": "Armada",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/crossovers---suvs/armada",
        "children": [],
        "Title": "Armada",
        "MetaTagDescription": "Armada Car"
      },
      {
        "id": 131,
        "name": "Juke",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/crossovers---suvs/juke",
        "children": [],
        "Title": "Juke",
        "MetaTagDescription": "Juke Car"
      },
      {
        "id": 132,
        "name": "Kicks",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/crossovers---suvs/kicks",
        "children": [],
        "Title": "Kicks",
        "MetaTagDescription": "Kicks Car"
      },
      {
        "id": 133,
        "name": "Murano",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/crossovers---suvs/murano",
        "children": [],
        "Title": "Murano",
        "MetaTagDescription": "Murano Car"
      },
      {
        "id": 134,
        "name": "Pathfinder",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/crossovers---suvs/pathfinder",
        "children": [],
        "Title": "Pathfinder",
        "MetaTagDescription": "Pathfinder Car"
      },
      {
        "id": 151,
        "name": "Quest",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/crossovers---suvs/quest",
        "children": [],
        "Title": "Quest",
        "MetaTagDescription": "Quest Cars"
      },
      {
        "id": 152,
        "name": "Rogue",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/crossovers---suvs/rogue",
        "children": [],
        "Title": "Rogue",
        "MetaTagDescription": "Rogue Cars"
      },
      {
        "id": 153,
        "name": "Rogue Sport",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/crossovers---suvs/rogue-sport",
        "children": [],
        "Title": "Rogue Sport",
        "MetaTagDescription": "Rogue Sport Cars"
      },
      {
        "id": 154,
        "name": "Xterra",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/crossovers---suvs/xterra",
        "children": [],
        "Title": "Xterra",
        "MetaTagDescription": "Xterra Cars"
      }
    ],
    "Title": "CROSSOVERS & SUVS",
    "MetaTagDescription": "CROSSOVERS & SUVS Cars"
  },
  {
    "id": 135,
    "name": "Car Parts",
    "hasChildren": true,
    "url": "https://echidna.vtexcommercestable.com.br/car-parts",
    "children": [
      {
        "id": 136,
        "name": "Car Part Accessories",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/car-part-accessories",
        "children": [],
        "Title": "Car Part Accessories",
        "MetaTagDescription": "Car Part Accessories of Cars"
      },
      {
        "id": 137,
        "name": "Body, Interior and Trim",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/body-interior-and-trim",
        "children": [],
        "Title": "Body-Interior-and-Trim",
        "MetaTagDescription": "Body, Interior and Trim of Cars"
      },
      {
        "id": 138,
        "name": "Brakes",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/brakes",
        "children": [],
        "Title": "Brakes",
        "MetaTagDescription": "Brakes of Cars"
      },
      {
        "id": 139,
        "name": "Cooling System",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/cooling-system",
        "children": [],
        "Title": "Cooling System",
        "MetaTagDescription": "Cooling System of Cars"
      },
      {
        "id": 140,
        "name": "Electrical System",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/electrical-system",
        "children": [],
        "Title": "Electrical-System",
        "MetaTagDescription": "Electrical System of Cars"
      },
      {
        "id": 141,
        "name": "Engine",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/engine",
        "children": [],
        "Title": "Engine",
        "MetaTagDescription": "Engine of Cars"
      },
      {
        "id": 142,
        "name": "EV System",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/ev-system",
        "children": [],
        "Title": "EV-System",
        "MetaTagDescription": "EV System of Cars"
      },
      {
        "id": 143,
        "name": "Exhaust System",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/exhaust-system",
        "children": [],
        "Title": "Exhaust System",
        "MetaTagDescription": "Exhaust System of Cars"
      },
      {
        "id": 144,
        "name": "Fuel System",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/fuel-system",
        "children": [],
        "Title": "Fuel-System",
        "MetaTagDescription": "Fuel System of Cars"
      },
      {
        "id": 145,
        "name": "HVAC",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/hvac",
        "children": [],
        "Title": "HVAC",
        "MetaTagDescription": "HVAC"
      },
      {
        "id": 146,
        "name": "Nissan Merchandise",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/nissan-merchandise",
        "children": [],
        "Title": "Nissan-Merchandise",
        "MetaTagDescription": "Nissan Merchandise"
      },
      {
        "id": 147,
        "name": "Other",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/other",
        "children": [],
        "Title": "Other",
        "MetaTagDescription": "Other"
      },
      {
        "id": 148,
        "name": "Suspension and Steering",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/suspension-and-steering",
        "children": [],
        "Title": "Suspension-and-Steering",
        "MetaTagDescription": "Suspension and Steering of Cars"
      },
      {
        "id": 149,
        "name": "Transmission and Driveline",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/transmission-and-driveline",
        "children": [],
        "Title": "Transmission-and-Driveline",
        "MetaTagDescription": "Transmission and Driveline"
      },
      {
        "id": 150,
        "name": "Batteries",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/car-parts/batteries",
        "children": [],
        "Title": "Batteries",
        "MetaTagDescription": "Batteries of Cars"
      }
    ],
    "Title": "Car Parts",
    "MetaTagDescription": "Car Parts"
  },
  {
    "id": 167,
    "name": "Accessories",
    "hasChildren": true,
    "url": "https://echidna.vtexcommercestable.com.br/accessories",
    "children": [
      {
        "id": 168,
        "name": "Engine Accessories",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/accessories/engine-accessories",
        "children": [],
        "Title": "engine accessories",
        "MetaTagDescription": "https://cdn.zeplin.io/5fbf567269bf8798be530f65/assets/4183CF74-EC66-4EEB-BC8E-C38F0E16A44B.png"
      },
      {
        "id": 169,
        "name": "Exterior Accessories",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/accessories/exterior-accessories",
        "children": [],
        "Title": "Exterior Accessories",
        "MetaTagDescription": "https://cdn.zeplin.io/5fbf567269bf8798be530f65/assets/93AC9544-006D-4097-AC09-8C9DB66D0EC5.png"
      },
      {
        "id": 170,
        "name": "AC/Electronics Accessories",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/accessories/ac-electronics-accessories",
        "children": [],
        "Title": "AC/Electronics Accessories",
        "MetaTagDescription": "https://cdn.zeplin.io/5fbf567269bf8798be530f65/assets/475473F1-F996-4D5D-87F7-CA8A51D74848.png"
      },
      {
        "id": 171,
        "name": "Interior Accessories",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/accessories/interior-accessories",
        "children": [],
        "Title": "Interior Accessories",
        "MetaTagDescription": "https://cdn.zeplin.io/5fbf567269bf8798be530f65/assets/C97E7601-86BC-4AD0-9AFD-5926FA61F269.png"
      },
      {
        "id": 172,
        "name": "NISMO",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/accessories/nismo",
        "children": [],
        "Title": "NISMO",
        "MetaTagDescription": "NISMO"
      },
      {
        "id": 173,
        "name": "Truck and Towing",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/accessories/truck-and-towing",
        "children": [],
        "Title": "Truck-and-Towing",
        "MetaTagDescription": "Truck and Towing"
      }
    ],
    "Title": "Accessories",
    "MetaTagDescription": "Accessories"
  },
  {
    "id": 175,
    "name": "Merchandise",
    "hasChildren": true,
    "url": "https://echidna.vtexcommercestable.com.br/merchandise",
    "children": [
      {
        "id": 176,
        "name": "Apparel",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/merchandise/apparel",
        "children": [],
        "Title": "Apparel",
        "MetaTagDescription": "Apparel"
      },
      {
        "id": 177,
        "name": "Auto Accessories",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/merchandise/auto-accessories",
        "children": [],
        "Title": "Auto-Accessories",
        "MetaTagDescription": "Auto Accessories"
      },
      {
        "id": 178,
        "name": "Business Accessories",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/merchandise/business-accessories",
        "children": [],
        "Title": "Business Accessories",
        "MetaTagDescription": "Business Accessories"
      },
      {
        "id": 179,
        "name": "Hats",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/merchandise/hats",
        "children": [],
        "Title": "Hats",
        "MetaTagDescription": "Hats"
      },
      {
        "id": 180,
        "name": "Heritage Collection",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/merchandise/heritage-collection",
        "children": [],
        "Title": "Heritage-Collection",
        "MetaTagDescription": "Heritage Collection"
      },
      {
        "id": 181,
        "name": "Nismo Apparel",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/merchandise/nismo-apparel",
        "children": [],
        "Title": "Nismo-Apparel",
        "MetaTagDescription": "Nismo Apparel"
      },
      {
        "id": 182,
        "name": "Outerwear",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/merchandise/outerwear",
        "children": [],
        "Title": "Outerwear",
        "MetaTagDescription": "Outerwear"
      },
      {
        "id": 183,
        "name": "Sports and Leisure",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/merchandise/sports-and-leisure",
        "children": [],
        "Title": "Sports-and-Leisure",
        "MetaTagDescription": "Sports and Leisure"
      },
      {
        "id": 184,
        "name": "Titan",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/merchandise/titan",
        "children": [],
        "Title": "Titan",
        "MetaTagDescription": "Titan"
      },
      {
        "id": 185,
        "name": "Z",
        "hasChildren": false,
        "url": "https://echidna.vtexcommercestable.com.br/merchandise/z",
        "children": [],
        "Title": "Z",
        "MetaTagDescription": "Z"
      }
    ],
    "Title": "Merchandise",
    "MetaTagDescription": "Merchandise"
  }
]