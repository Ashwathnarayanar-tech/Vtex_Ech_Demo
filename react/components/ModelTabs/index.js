import React from 'react';
import Tabs from "./Tablist/Tabs";
import styles from './model.css';
import nissan_placeholder_image from "../../../assets/jellybean-placeholder.png";

const modelData = [
  {
    label: "POPULAR MODELS",
    contentType: "car",
    content: [
      {url: "javascript:void(0)", modelName: "Altima"},
      {url: "javascript:void(0)", modelName: "Leaf"},
      {url: "javascript:void(0)", modelName: "Maxima"},
      {url: "javascript:void(0)", modelName: "Rogue"},
      {url: "javascript:void(0)", modelName: "Sentra"},
      {url: "javascript:void(0)", modelName: "Titan"}
    ]
  },
  {
    label: "CARS",
    contentType: "car",
    content: [
      {url: "javascript:void(0)", modelName: "Altima"},
      {url: "javascript:void(0)", modelName: "Leaf"},
      {url: "javascript:void(0)", modelName: "Maxima"},
      {url: "javascript:void(0)", modelName: "Versa"},
      {url: "javascript:void(0)", modelName: "Sentra"},
      {url: "javascript:void(0)", modelName: "Versa Note"}
    ]
  },
  {
    label: "CLASSICS",
    contentType: "car",
    content: [
      {url: "javascript:void(0)", modelName: "200SX"},
      {url: "javascript:void(0)", modelName: "240SX"},
      {url: "javascript:void(0)", modelName: "240Z"},
      {url: "javascript:void(0)", modelName: "260Z"},
      {url: "javascript:void(0)", modelName: "280Z"},
      {url: "javascript:void(0)", modelName: "280ZX"},
      {url: "javascript:void(0)", modelName: "300ZX"},
      {url: "javascript:void(0)", modelName: "310"},
      {url: "javascript:void(0)", modelName: "310GX"},
      {url: "javascript:void(0)", modelName: "350Z"},
      {url: "javascript:void(0)", modelName: "410"},
      {url: "javascript:void(0)", modelName: "411"}
    ]
  },
  {
    label: "COMMERCIAL",
    contentType: "car",
    content: [
      {url: "javascript:void(0)", modelName: "Frontier"},
      {url: "javascript:void(0)", modelName: "NV1500"},
      {url: "javascript:void(0)", modelName: "NV200 Taxi"},
      {url: "javascript:void(0)", modelName: "NV2500"},
      {url: "javascript:void(0)", modelName: "NV3500"},
      {url: "javascript:void(0)", modelName: "Titan"}
    ]
  },
  {
    label: "CROSSOVERS & SUVS",
    contentType: "car",
    content: [
      {url: "javascript:void(0)", modelName: "Armada"},
      {url: "javascript:void(0)", modelName: "Juke"},
      {url: "javascript:void(0)", modelName: "Kicks"},
      {url: "javascript:void(0)", modelName: "Murano"},
      {url: "javascript:void(0)", modelName: "Path Finder"},
      {url: "javascript:void(0)", modelName: "Quest"},
      {url: "javascript:void(0)", modelName: "Rogue"},
      {url: "javascript:void(0)", modelName: "Rogue Sport"}
    ]
  },
  {
    label: "ELECTRIC",
    contentType: "date",
    content: [
      {url: "javascript:void(0)", modelName: "2021"},
      {url: "javascript:void(0)", modelName: "2020"},
      {url: "javascript:void(0)", modelName: "2019"},
      {url: "javascript:void(0)", modelName: "2018"},
      {url: "javascript:void(0)", modelName: "2017"},
      {url: "javascript:void(0)", modelName: "2016"}
    ]
  },
  {
    label: "SPORTS CARS",
    contentType: "car",
    content: [
      {url: "javascript:void(0)", modelName: "370Z"},
      {url: "javascript:void(0)", modelName: "GT-R"}
    ]
  },
  {
    label: "TRUCKS",
    contentType: "car",
    content: [
      {url: "javascript:void(0)", modelName: "Frontier"},
      {url: "javascript:void(0)", modelName: "Titan"}
    ]
  }
];

function ModelTabs() {
  return (
    <div className={styles.modelTabWrapper}>
      <Tabs>
      {modelData.map((child) => {
        const type = child.contentType;
        return (
          <div label={child.label} className={styles.type}>
            <div className={styles.tabContentContainer}>
              <div className={styles.carModelContainer}>
                <ul className={styles.modelRange}>
                {child.content.map((contents) => {
                  return (
                    <li className={styles.modelList}>
                      <a href={contents.url} className={styles.modelAnchor}>
                        <img className={styles.carImage} src={nissan_placeholder_image} alt={contents.modelName} />
                        <h4 className={styles.modelTitle}>{contents.modelName}</h4>
                      </a>
                    </li>
                  )
                })}
                </ul>
              </div>
            </div>
          </div>
        );
      })}
      </Tabs>
    </div>
  );
}

export default ModelTabs;