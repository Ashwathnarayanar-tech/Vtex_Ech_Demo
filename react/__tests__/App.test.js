import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import ModelTabs from "../components/ModelTabs/index";
import ContactUsData from "../components/ContactUsData";
import AddressBanner from "../components/AddressBanner";
import { makeAPICall } from "httpCall";
import { expectedResponse } from "../__mocks__/response";

// Configuration
configure({ adapter: new Adapter() });

// To check Tabs component is rendered properly inside ModelTabs component
describe("ModelTabs", () => {
  test("Tabs component rendered properly", () => {
    const component = shallow(<ModelTabs />);
    const TabsExists = component.find("Tabs").exists();
    expect(TabsExists).toBe(true);
  });
});

// AddressBanner Component
describe("AddressBanner", () => {
  // To find the text EMAIL US
  it("should show text Email Us", () => {
    const wrapper = shallow(<AddressBanner />);
    const text = wrapper.find("div div div a");
    expect(text.text()).toBe("EMAIL US");
  });
  // To find the text RETURN POLICY
  it("should show text RETURN POLICY", () => {
    const wrapper = shallow(<AddressBanner />);
    const text = wrapper.find("div div a h4");
    expect(text.text()).toBe("RETURN POLICY");
  });
 // Snapshot test
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<AddressBanner debug />);
    expect(component).toMatchSnapshot();
  });
  // props testing (text)
  it("Props rendered properly", () => {
    const wrap = shallow(<AddressBanner name="View" />);
    expect(wrap.text()).toEqual(
      "ViewEMAIL USPolicies & InformationRETURN POLICYACCEPTED PAYMENTS"
    );
  });
});

// ============================================================

describe("ContactUsData", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ContactUsData debug />);

    expect(component).toMatchSnapshot();
  });
});
// Api testing

describe("API test", () => {
  it("Api testing", async function () {
    const response = new makeAPICall();
    // console.warn(await response);
    expect(await response).toEqual(expectedResponse);
  });
});
