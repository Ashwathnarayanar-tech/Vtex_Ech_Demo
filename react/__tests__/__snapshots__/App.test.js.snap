// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressBanner should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`ContactUsData should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
